src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,133 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import React, { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [searchParams] = useSearchParams();
+  const Male = 'm';
+  const Female = 'f';
+
+  const preparedPeople = people.map(personValue => ({
+    ...personValue,
+    mother: people.find(human => human.name === personValue.motherName),
+    father: people.find(human => human.name === personValue.fatherName),
+  }));
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(setPeople)
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handleFilter = () => {
+    let filteredPeople = [...preparedPeople];
+    const query = searchParams.get('query') || '';
+    const centuries = searchParams.getAll('centuries') || [];
+    const sex = searchParams.get('sex') || '';
```

Comment:
Get all query params at the top of the component, usually the flow look like this.
```typescript
// Declare all states
const [people, setPeople] = useState<Person[]>([]);
// other states

// Declare all variables
const query = searchParams.get('query') || '';
const centuries = searchParams.getAll('centuries') || [];
const sex = searchParams.get('sex') || '';

// Prepare data
const filteredPeople = filterPeople(people, { query, sex, centuries });
const sortedPeople = sortPeople(people, sortParams);

// Effects

// handlers

// return jsx
```